lib/components/DynamicFonts/index.ts

```diff
@@ -145,11 +139,31 @@ export default class DynamicFonts {
         fontsLoaded = await this.getFonts(fonts);
       } catch (error) {
         if (debug) {
-          console.log(`getFontFamily failed (try #${tryCounter}) error: ${error}`);
+          console.log(`getFontFamily failed (try #${tryCounter}) error:`, error);
         }
       }
     }
 
     return Promise.resolve(fontsLoaded);
   }
+
+  private async deleteFontFromDisk(fontName: string, fontExtension: FontExtension, fontNamePrefix?: string) {
+    const fontInput = this.buildFontInput('', fontName, fontExtension, fontNamePrefix);
+    await this.fontDownloader.deleteFontFromDisk(fontInput.fullFontName);
+  }
+
+  public async deleteFont(fontName: string, fontExtension: FontExtension): Promise<void> {
+    await this.permissionsAcquirer.getPermissions();
+    if (await this.fontDownloader.isFontDownloaded(fontName, fontExtension)) {
+      await this.deleteFontFromDisk(fontName, fontExtension);
+    }
+  }
+
+  public async deleteFontFamily(fontNames: string[],
+    fontExtension: FontExtension,
+    fontNamePrefix?: string): Promise<void> {
+    fontNames.forEach(async fontName => {
+      await this.deleteFontFromDisk(fontName, fontExtension, fontNamePrefix);
```

Comment:
Here you're not checking if 'isFontDownloaded' before deleting it?

Reply:
I'm thinking of removing this test altogether since a non-existing is checked for later anyway, WDYT?

Reply:
If it's checked later, yes remove it

Reply:
Removed, but added a permission check